features/built_in_matchers/match_pattern.feature

```diff
@@ -0,0 +1,27 @@
+Feature: `match_pattern` matcher
+
+  Use the `match_pattern` matcher to specify that a value matches with
+  expected patterns by Ruby's pattern-matching.
+
+    ```ruby
+    expect([1, 2, 3]).to match_pattern([Integer, Integer, Integer])
+    expect([1, 2, 3]).to match_pattern([Integer, Integer, String])
+    ```
+
+  Scenario: Pattern usage
```

Comment:
This would also have to be tagged and excluded on unsupported ruby, and as it is used in our docs we would need a note about supported ruby versions in that,